third_party/terraform/resources/resource_composer_environment.go.erb

```diff
@@ -932,6 +1043,38 @@ func expandComposerEnvironmentConfigWebServerNetworkAccessControl(v interface{},
 }
 
 <% end -%>
+
+<% unless version == "ga" -%>
+func expandComposerEnvironmentConfigDatabaseConfig(v interface{}, d *schema.ResourceData, config *Config) (*composer.DatabaseConfig, error) {
+	l := v.([]interface{})
+	if len(l) == 0 {
+		return nil, nil
+	}
+	raw := l[0]
+	original := raw.(map[string]interface{})
+
+	transformed := &composer.DatabaseConfig{}
+	transformed.MachineType = original["machine_type"].(string)
+
+	return transformed, nil
+}
+
+func expandComposerEnvironmentConfigWebServerConfig(v interface{}, d *schema.ResourceData, config *Config) (*composer.WebServerConfig, error) {
+	l := v.([]interface{})
+	if len(l) == 0 {
```

Comment:
This file has some fairly old expanders in it, and they don't handle the `[]interface{}{nil}` case I mentioned above. We can safeguard against it by making the sub-fields `Required`, but we can also steal some code from a more up-to-date file too. https://github.com/hashicorp/terraform-provider-google/blob/master/google/resource_compute_health_check.go#L1470-L1475 is where I got this from.

```suggestion
	if len(l) == 0 || l[0] == nil{
```